### ISSUE TITLE
S2I scripts should be used for build/run

### ISSUE THREAD

**Issue by Ladicek** [halkyonio/operator#178]
The `quay.io/halkyonio/hal-maven-jdk` image uses has its own `build` and `run` scripts, which are quite simplistic. Building a Maven project in a generic fashion, as well as running an uberjar in a container environment, are not that easy. Fortunately, the problems have already been solved in the S2I Java images, which have the `assemble` and `run` scripts. These scripts contain a lot of logic that should be reused.

Related to that, Halkyon uses some filesystem paths that are for some reason different from the S2I paths:

S2I images / scripts | Halkyon images / scripts
------------ | -------------
`/tmp/src` | `/usr/src`
`/tmp/artifacts` | `/tmp/artefacts`

From my cursory understanding of Halkyon, this is a combination of typos, workarounds for permission issues, and NIH.

IMHO, existing logic should be reused as much as possible.

**Comment by metacosm** [comment]
I agree, this just hasn't been a priority so far. PRs are always considered 😄
